service/models.py

```diff
@@ -86,18 +85,20 @@ def delete(cls, supplier_id):
         """ Delete a supplier by it's ID """
         cls.logger.info('Processing deleting for id %s', supplier_id)
         try:
-            return cls.objects(id=supplier_id).delete()
-        except DoesNotExist:
+            res = cls.objects(id=supplier_id).first()
+        except ValidationError:
             return None
```

Comment:
For the following testing, let's raise a self-defined DataValidationError instead of return None:
```
except DoesNotExist as error:
            raise DataValidationError('Invalid supplier_id: failed to '
                'find a supplier with given supplier_id')
```